examples/cvector-generator/mini-tests/test-vanilla-pca.cpp

```diff
@@ -0,0 +1,116 @@
+
+#include "common.h"
+#include "llama.h"
+#include "ggml.h"
+#include "../pca.hpp"
+
+#ifdef GGML_USE_CUDA
+#include "ggml-cuda.h"
+#endif
+
+#ifdef GGML_USE_METAL
+#include "ggml-metal.h"
+#endif
```

Comment:
These includes are not necessary. Include `ggml-cpp.h` for the smart pointers, and `ggml-backend.h` for the backend funtions.

Reply:
Okayy!